### ISSUE TITLE
[Potential NAN bug] Loss may become NAN during training even if 1e-10 was added in log()

### ISSUE THREAD

**Issue by Justobe** [iusaspb/deep_learning_Nikolenko_and_Co#1]
Hello~

Thank you very much for sharing your code.

I ran ch10_04_01.py on my computer. Unfortunately, I got NAN loss. 

![image-20201103114457101](https://user-images.githubusercontent.com/15032308/97980958-87b29280-1e0c-11eb-9394-117e339d313d.png)

After preliminary inspection of the code, I couldn't find the root cause of NaN loss. 1e-10 was added to the log  to prevent log(0). This confused me for a long time. After I checked the program carefully with tfdebug, I found the problem!  

In the code, 1e-10 was added to log to prevent log(0)

```python
reconstr_loss = -tf.reduce_sum(x * tf.log(1e-10 + x_reconstr_mean) +(1-x) * tf.log(1e-10 + 1-x_reconstr_mean), 1)
```

However, 1e-10 may be too small to avoid `1e-10 + 1-x_reconstr_mean` becoming 0, because I found that x_reconstr_mean may sometimes become around 1.0000000001 due to floating-point accuracy issue, which may lead to log(0) 

Increasing 1e-10 to 1e-7 can solve this problem.

**Comment by iusaspb** [comment]
Thank you.
